src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs

```diff
@@ -25,6 +27,28 @@ public static void EnsureDotnet()
             }
         }
 
+        /// <summary>
+        /// Function that determines TargetFramework of a project even when it's defined outside of the .csproj file,
+        /// e.g. in Directory.Build.props
+        /// </summary>
+        /// <param name="projectDirectory">Directory containing the .csproj file</param>
+        /// <returns>Target framework, e.g. net8.0</returns>
+        /// <exception cref="CliException"></exception>
+        public static async Task<string> DetermineTargetFramework(string projectDirectory)
+        {
+            EnsureDotnet();
+            var exe = new Executable("dotnet", "build -getproperty:TargetFramework", workingDirectory: projectDirectory);
```

Comment:
Thanks @gukoff for the contribution.

The change looks good. Would you mind addressing  [this comment](https://github.com/Azure/azure-functions-core-tools/pull/3604#issuecomment-2221428606) by @jviau?  I think no-logo can be passed in the command itself `dotnet build -getproperty:TargetFramework --nologo` . The other 2,we may need to set via env variables for this command (and may need to reset to previous state after this command is executed)

Reply:
Good idea @jviau, I added the env vars.

One caveat: I haven't worked on this project for a while, and I don't remember how to test these changes :(